hud/powerdisplay.lua

```diff
@@ -146,17 +169,24 @@ function powerDisplay.widget(glasses, data)
     end
 end
 
-function powerDisplay.remove()
+function powerDisplay.remove(glassAddress)
     for i = 1, #hudObjects do
-        for j = 1, #hudObjects[i].static do
-            hudObjects[i].glasses.removeObject(hudObjects[i].static[j].getID())
+        local hudObject = hudObjects[i]
+        local glasses = hudObject.glasses
+        if glasses ~= nil then
+            if glasses.address == glassAddress then
+                for j = 1, #hudObjects[i].static do
+                    hudObjects[i].glasses.removeObject(hudObjects[i].static[j].getID())
+                end
+                hudObjects[i].glasses.removeObject(hudObjects[i].dynamic.energyBar.getID())
+                hudObjects[i].glasses.removeObject(hudObjects[i].dynamic.currentEU.getID())
+                hudObjects[i].glasses.removeObject(hudObjects[i].dynamic.maxEU.getID())
+                hudObjects[i].glasses.removeObject(hudObjects[i].dynamic.percentage.getID())
+                hudObjects[i].glasses.removeObject(hudObjects[i].dynamic.filltime.getID())
+                hudObjects[i].glasses.removeObject(hudObjects[i].dynamic.fillrate.getID())
+                hudObjects[i].glasses.removeObject(hudObjects[i].dynamic.state.getID())
```

Comment:
Could you iterate through the `dynamic` table?

Reply:
Maybe, but I don't know how to do that off the top of my head. Maybe with pairs()?

Reply:
Yes, probably